Donation-platform/src/main/webapp/browse.js

```diff
@@ -105,50 +40,46 @@ function getCharitiesForDisplay(tagName)
 /**
  * Populate browse page with category cards.
  */
-function populateBrowsePage(charityCollection)
+function populateBrowsePage(tagCollection)
 {
     document.getElementById('pagination').innerHTML = '';
-    var tagCharityDict = tagToCharityDictMapper(charityCollection);
-    var orderedTagCharityDict = sortDict(tagCharityDict);
-    var tagImageDict = tagToImageSourceMapper(charityCollection);
     // Clears previous inner HTML data to prevent duplicate data
     // from being shown on the webpage.
     document.getElementById('collection').innerHTML = "";
     const cards = document.getElementById('collection');
-    var charityDivElement;
-    Object.keys(orderedTagCharityDict).forEach(function (tag)
+    var tagDivElement;
+    Object.keys(tagCollection.sort(compare)).forEach(function (tag)
     {
-        charityDivElement = document.createElement( "div" );
-        charityDivElement.setAttribute("class", "charity-card");
-        var charityDivInternalElement = document.createElement( "div" );
-        charityDivInternalElement.setAttribute("class", "card charity-card-internal");
-        var charityImgElement = document.createElement( "img" );
-        charityImgElement.setAttribute("id", "card-img");
-        charityImgElement.setAttribute("class", "card-img-top");
-        charityImgElement.setAttribute("src", tagImageDict[tag]);
-        charityImgElement.setAttribute("alt", "Card image" );
-        var charityDivBodyElement = document.createElement( "div" );
-        charityDivBodyElement.setAttribute("class", "card-body d-flex flex-column");
-        var charityHeaderElement = document.createElement( "h4" );
-        charityHeaderElement.setAttribute("class", "card-title");
-        var headerText = document.createTextNode(tag.charAt(0).toUpperCase() + tag.slice(1));
-        charityHeaderElement.appendChild(headerText);
-        var seeCharityButtonElement = document.createElement( "button" );
-        seeCharityButtonElement.setAttribute("class", "mt-auto btn btn-primary");
-        seeCharityButtonElement.setAttribute("id", "charity-btn");
-        var seeCharityText = document.createTextNode("See Charities");
-        seeCharityButtonElement.appendChild(seeCharityText);
-        seeCharityButtonElement.addEventListener("click", function() {
-            getCharitiesForDisplay(tag);
+        tagDivElement = document.createElement( "div" );
+        tagDivElement.setAttribute("class", "charity-card");
+        var tagDivInternalElement = document.createElement( "div" );
+        tagDivInternalElement.setAttribute("class", "card charity-card-internal");
```

Comment:
split this into a few helper functions for setting the fields in each of the div, img, header elements etc.

Reply:
will be fixed in next PR.